File: problems/0008_string-to-integer-atoi/built-in/Solution.swift
class Solution {
    func myAtoi(_ str: String) -> Int {
        if let match = NUM_FMT.firstMatch(
                in: str, range: NSRange(location: 0, length: str.count)) {
            if let r = Range(match.range(at: 1), in: str) {
                if let result = Decimal(string: String(str[r])) {
                    if result < INT32_MIN_DEC {
                        return INT32_MIN_INT
                    } else if result > INT32_MAX_DEC {
                        return INT32_MAX_INT
                    } else {
                        return Int(NSDecimalNumber(decimal: result))
                    }
                }
            }
        }
        return 0
    }
    let (INT32_MIN_DEC, INT32_MAX_DEC,
         INT32_MIN_INT, INT32_MAX_INT,
         NUM_FMT) =
            (Decimal(Int32.min), Decimal(Int32.max),
             Int(Int32.min), Int(Int32.max),
             try! NSRegularExpression(pattern: "^\\s*([\\-\\+]?\\d+).*$"))
}

File: problems/0008_string-to-integer-atoi/built-in/Solution.php
<?php
class Solution {
    /**
     * @param String $str
     * @return Integer
     */
    function myAtoi($str) {
        if (preg_match('/^\s*([\-\+]?\d+).*$/', $str, $matches)) {
            $result = intval($matches[1]);
            if ($result < -pow(2, 31))
                return -pow(2, 31);
            else if ($result > pow(2, 31) - 1)
                return pow(2, 31) - 1;
            else
                return $result;
        }
        return 0;
    }
}
